### PR TITLE
add cache test

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This PR implements internal/cache test case.

### Environment:

- Go Version: 1.14.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.0

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by vankichi** [comment]
@kevindiu  please confirm description and others required.

File: internal/cache/cache_test.go [hunk 1]
```diff
@@ -52,36 +59,36 @@ func TestNew(t *testing.T) {
...
+			want: want{
+				err: errors.ErrInvalidCacherType,
+			},
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(tt *testing.T) {
-			defer goleak.VerifyNone(t)
+			defer goleak.VerifyNone(t, goleakIgnoreOptions...)
```
Comment (vankichi) [inline internal/cache/cache_test.go:91]:
```suggestion
			defer goleak.VerifyNone(tt, goleakIgnoreOptions...)
```

**Comment by vankichi** [comment]
Could you add new case related [this](https://codecov.io/gh/vdaas/vald/pull/576/changes#D1-62)?

**Comment by vankichi** [comment]
@kevindiu did you use old template?
I think we should use the latest template. @hlts2 how do you think about it ?

**Comment by kevindiu** [comment]
> Could you add new case related [this](https://codecov.io/gh/vdaas/vald/pull/576/changes#D1-62)?

I think it is untestable as the default options contains gache type and the `WithType()` blocks unknown type...

**Comment by kevindiu** [comment]
> @kevindiu did you use old template?
> I think we should use the latest template. @hlts2 how do you think about it ?

yes I didn't re-generate the test code so i think it is using the old one.

**Comment by kevindiu** [comment]
I will regenerate the test code later

File: internal/cache/option.go [hunk 2]
```diff
@@ -36,52 +35,45 @@ var (
...
 			c.expiredHook = f
 		}
-		return nil
 	}
 }
 
 func WithType(mo string) Option {
-	return func(c *cache) error {
+	return func(c *cache) {
 		if len(mo) == 0 {
```
Comment (kevindiu) [inline internal/cache/option.go:47]: @hlts2 which one do you think is better?
```suggestion
		if mo != "" {
```

Comment (hlts2) [reply internal/cache/option.go:47]: I think `len(mo) == 0` is better.

Comment (hlts2) [reply internal/cache/option.go:47]: I'm not particular about it, so I think either one is fine.
but uber use `len() == 0` in to the `[style guide](https://github.com/uber-go/guide/blob/master/style.md#dont-panic)

Comment (kevindiu) [reply internal/cache/option.go:47]: thank you.

**Pull request merged**